### DISCUSSION TITLE
Sticky top header

### DISCUSSION THREAD

**Discussion by balrogmedia** [bootscore/bootscore#288] [Q&A] [answered]
Hello,

I'm having some issues with the sticky-top class.

when I scroll down, the <div> sticks for a little bit, but then it scrolls up anyway.

Is this a recurrent issue or is it just me? any idea how to go around it?

thank you

**Comment by crftwrk** [comment]
Hello,

a sticky element needs the space to be sticky. Check this discussion here https://github.com/bootscore/bootscore/discussions/217.

To help you out here, please explain what you want to do in detail and of course, a link to your project is always very helpful.

**Comment by balrogmedia** [comment]
hi Basti, thank you for your kind reply

https://casinoonlinesoldiveri.it/

if you take a look on my site, there is an image on top the navbar, which has to be there for compliance reasons. when you scroll down, the image correctly disappear and the navbar sticks for a little bit. But if you keep scrolling it also disappears. I'd like for the navbar to stick.

**Reply by crftwrk** [reply]
`sticky-top` won't work there, because header is at same level as content. Means if you reached the part of the site content, element cannot be sticky anymore. That's exactly the same case as described here  https://github.com/bootscore/bootscore/discussions/217 and that's why header has `fixed-top` class by default. 

In your case I suggest to check if header is on top and add `fixed-top` class then.

First remove `sticky-top` class in `header.php`:

```HTML
<header id="masthead" class="site-header">
```

Then add following snippet to your child `js/custom.js` inside the `jQuery(function ($) {`...`});` wrapper:

```JS
var distance = $('#masthead').offset().top,
  $window = $(window);

$window.scroll(function () {
  if ($window.scrollTop() >= distance) {
    $(#masthead').addClass('fixed-top');
  } else {
    $('#masthead').removeClass('fixed-top');
  }
});

```

Clear browser cache.

Does this help?

**Comment by balrogmedia** [comment]
wow, thats great!

weirdly it works from logged into WP, but not from logged out for some reasons, even if I cleaned browser and kinsta cache. Any idea why?

thank you

**Comment by crftwrk** [comment]
Did you add the jQuery snippet? Your `custom.js` is empty? Do you have any caching plugin active?

![screenshot](https://user-images.githubusercontent.com/51531217/195422088-3156ea07-90e6-46f7-9305-7784ae3cda1f.png)

**Comment by balrogmedia** [comment]
I did but, and the script works when I am logged into WP. From logged out it doesn't seem be updated.

**Reply by crftwrk** [reply]
Please add snippet again. Then I can check it as a logged-out user.

**Comment by balrogmedia** [comment]
Good morning, I've done it again but it doesn't work from logged out. I guess I will just gave up on it.

One more question about BootScore, is it possible from desktop to swap the search form for the search icon like in mobile? it would be much more elegant

thank you

**Reply by crftwrk** [reply]
There is no difference between logged-in and logged-out. What I can say that script isn't there. This can be if you have a caching plugin active which caches for users but not for admins, for example Autoptimize. You can create me an admin account and I will check it again, it should not be a big deal.

> One more question about BootScore, is it possible from desktop to swap the search form for the search icon like in mobile? it would be much more elegant

Of course, this is possible. But please open a new discussion for this to keep issues organised.

**Comment by balrogmedia** [comment]
ok thanks,

here is the credentials:
ID: bootscore
PW: xDB4ITuh8#lUKDLkBsymmvrg
https://casinoonlinesoldiveri.it/wp-admin/

**Comment by crftwrk** [comment]
Ok, this is really strange. As I wrote before, it is a plugin that caches something. But it is not as expected WP Rocket, it is Sucuri Security - Auditing, Malware Scanner and Hardening. Maybe you can try https://wordpress.org/plugins/better-wp-security/ instead.

However, you changed code in header.php. I adapted jQuery snippet to this. Now it works for logged-out users, but not for logged-in users. This is because one of plugins mentioned above caches different for users and admins. As logged-in, it's the old code, logged-out the new working one 🤪

So, you have to go through both plugin settings to solve this.

**Comment by balrogmedia** [comment]
like this works like a charm!

thank you very much :)

**Reply by crftwrk** [reply]
👍 Please mark this https://github.com/bootscore/bootscore/discussions/288#discussioncomment-3861569 as answered.

Thank you

**Comment by balrogmedia** [comment, marked as answer]
Thanks very much for all the help!

**Reply by crftwrk** [reply]
You're welcome. Please mark this https://github.com/bootscore/bootscore/discussions/288#discussioncomment-3861569 as answered.

**Reply by balrogmedia** [reply]
how? im trying... but I dont see an option

**Reply by crftwrk** [reply]
Yes, sorry. Now ;-)

<img width="242" alt="Bildschirmfoto 2022-10-13 um 13 11 46" src="https://user-images.githubusercontent.com/51531217/195581781-22cb3852-3de3-43dd-a6aa-7111f81d093f.png">
